### PR TITLE
Bug/4592 fix new line padding class diagram

### LINKED ISSUE #4592: newline(s) in classDiagram class name causes all members to each have same newline(s)
### Description

When I put `\n` in a class diagram class name, this adds the same newlines to all class members below.
Easy example with mermaid from https://github.com/mermaid-js/mermaid/issues/3958#issuecomment-1474173500

### Result

The layout of the `Du\nck` class is doublespaced in the class header and throughout the class members. Resulting in the members being rendered outside the class box. And when a class has many members, it results in unreadable layout where members are over/under other class boxes.

### Expected

Class members spacing to be unrelated to the spacing/height of the class name. They are always inside their own class box.

### Steps to reproduce

1. Use example mermaid from https://github.com/mermaid-js/mermaid/issues/3958#issuecomment-1474173500
2. Paste it into https://mermaid.live/

### Screenshots

image

### Code Sample

```text
https://mermaid.live/edit#pako:eNpVksFunDAQhl8FzanVsiwYYgPqpWoSqYeccouQkgEbsBbsrW2abrb77rVpts2ePP78z3j-sU_QaS6ghm5Ca28lDgbnRn1VcsYp-vJ7u41ebpemUd3-5RrfSztekyfRGvyH6mgjlYtwEB_RozNSDdEgFBfmchCk9gFnH376_AHO6EQAa2__-4hOjboUagXuv-lJ-1ob-yrnIN_8WLDbh-h8yQ3N-qxt6MjKN_Fd3QvhPOhQ3aG70q42whWt1lMk7fOrnLjfmkWtOohhFmZGyf3YvC6KGnCjmEUDtQ-56HGZXAN_pbg4_XhUHdTOLCKG5cC9q_dBX8M7Lp02UPc4WQ8njX5GUJ_AHQ_hiQZpna_YadXLIfDFTB6Pzh1svduF42SQblzapNPzzko-onHjz4ruKKElklxQluNNnvOuzaqyJ0XWc5ZmBOF8juGAKlT9BXVWZEnBaFWRlGW0KKsihiPUhNCEUJpWOUtZwcqM-Kw3rb2PNGEVK2mVpyzPqoykNzGI1c_D-_8Ky3rH05qwmjz_AZ7P0sw
```

### Setup

- Mermaid version: 10.2.3
- Browser and Version: Chrome Version 114.0.5735.199 (Official Build) (64-bit)

also reproduces in VSCode with https://marketplace.visualstudio.com/items?itemName=bierner.markdown-mermaid

### Suggested Solutions

None known. One has to remove all `\n` from class names.

### Additional Context

Errant behavior occurs when using immediate `Du\nck` or the label method `["Du\nck"]`.
If the class has no members, then the errant behavior is silently not seen.

### PR DESCRIPTION
**Pull request by ibrahimWassouf**
## :bookmark_tabs: Summary

Changes how class member height adds to class diagram overall height

Resolves #4592 

## :straight_ruler: Design Decisions

Previously, for each class member the title height was added assuming the titleHeight was a single line. Now, we calculate the height of each member's Bounding Box and add it to the vertical position to render the class properly

### :clipboard: Tasks

Make sure you

- [x] :book: have read the [contribution guidelines](https://github.com/mermaid-js/mermaid/blob/develop/CONTRIBUTING.md)
- [x] :computer: have added unit/e2e tests (if appropriate)
- [ ] :notebook: have added documentation (if appropriate)
- [x] :bookmark: targeted `develop` branch

### CODE REVIEW

File: packages/mermaid/src/dagre-wrapper/nodes.js [hunk 1]
```diff
@@ -917,7 +917,9 @@ const class_box = (parent, node) => {
         ((-1 * maxHeight) / 2 + verticalPos + lineHeight / 2) +
         ')'
     );
-    verticalPos += classTitleBBox.height + rowPadding;
+    //get the height of the bounding box of each member if exists
+    const memberBBox = lbl ? lbl.getBBox() : null;
+    verticalPos += (memberBBox.height ?? 0) + rowPadding;
```
Comment (sidharthv96) [inline packages/mermaid/src/dagre-wrapper/nodes.js:922]:
```suggestion
    const memberBBox = lbl?.getBBox();
    verticalPos += (memberBBox?.height ?? 0) + rowPadding;
```

File: packages/mermaid/src/dagre-wrapper/nodes.js [hunk 2]
```diff
@@ -935,7 +937,8 @@ const class_box = (parent, node) => {
       'transform',
       'translate( ' + -maxWidth / 2 + ', ' + ((-1 * maxHeight) / 2 + verticalPos) + ')'
     );
-    verticalPos += classTitleBBox.height + rowPadding;
+    const methodBBox = lbl ? lbl.getBBox() : null;
+    verticalPos += (methodBBox.height ?? 0) + rowPadding;
```
Comment (sidharthv96) [inline packages/mermaid/src/dagre-wrapper/nodes.js:941]:
```suggestion
    const memberBBox = lbl?.getBBox();
    verticalPos += (memberBBox?.height ?? 0) + rowPadding;
```

**Review by sidharthv96 — CHANGES_REQUESTED** [review]

**Comment by ibrahimWassouf** [comment]
Thanks for the review @sidharthv96, I still got a lot to learn!

File: cypress/integration/rendering/classDiagram.spec.js [hunk 3]
```diff
@@ -423,4 +423,67 @@ describe('Class diagram', () => {
     );
     cy.get('svg');
   });
+
+  it('20: should render class diagram with newlines in title', () => {
```
Comment (sidharthv96) [inline cypress/integration/rendering/classDiagram.spec.js:427]: You can remove the numberings from the tests. It was an old format and no one's sure why it was added. 

Also, can you add tests for string labels? Or is it already present in the file?

```
classDiagram
    class Animal["Animal with a\nNewline"]
```

Comment (ibrahimWassouf) [reply cypress/integration/rendering/classDiagram.spec.js:427]: Should I also remove the two commented tests?

**Comment by sidharthv96** [comment]
@ibrahimWassouf I have reverted 80add648e6d82eb99508b7ffa2acbf52200905c9 as changes to old tests might affect the Applitools test suite. 
Thanks for making the changes in a separate commit :)

**Pull request merged**